src/core/Config.cpp

```diff
@@ -337,44 +337,55 @@ void Config::setDefaultActionKeys() {
 	}
 }
 
-void Config::setActionKey(ControlAction actionId, int index, InputKeyId key) {
+void Config::setActionKey(const ControlAction actionId, uint32_t index, InputKeyId newKey) {
```

Comment:
Sure, my comment was more about it being 32-bit. While `int` is also 32-bit in practice it does not mean that it *mus* be 32-bit.

Like, Eli2 commented on irc, size_t (or a signed int) should be used for indices. Smaller fixed sizes for indices only make sense as a performance optimization (not important here) and in this case it would only make things worse (see the video Eli2 liked in irc if you are interested in why).